### ISSUE TITLE
Lastingly remember the notes and color marks given? (And many more things?)

### ISSUE THREAD

**Issue by user0022** [madprops/grasshopper#178]
Hello again!

4 months after my last post in these issue pages, still using your addon everyday and very happy about it, so I want to say again thank you for it! Still seems to me so light (but I'm no specialist so maybe it's normal) when thinking of all its useful features.

*
Personnally I use colors a bit like a lasting or temporary mark to show important or currently important tabs.

They stay present on the tab even if the URL is changed, which is good for this use of page reminder as the URL can change partially (like with and internal anchor) or more, while this tab still containing the wanted page. 

But I have seen a case when they disappear. If the tab is closed, and the history is flushed (manually in about:preferences#privacy, or maybe if browsing with 0 history but I didn't test because I feared to loose my session). And then if the tab is repoened, the color is not there anymore. For me this could be a problem even if the case can be rare? If worth it, so could they be kept even in this case? Maybe by keeping the tab/color association in a file the history clearing doens't suppress, but I'm no coder!
(The problem doesn't happen when I flush only the cache.)

While I'm about color marks, I wanted also to remark that the color named "Yellow" is rather orange no?
If correcting, it could be either the color or the name that would be. Maybe orange color is a bit less visually aggressive than yellow, and more common so maybe keep the color. Or also maybe add the yellow as a 5th mark...

*
For the notes, in my tests it is  bit worse, as simply closing/reopening a tab and flushing history OR cache betweeen these 2 actions makes them disappear.
I had tested the color mark  4 or 5 months ago and it was the same (at least in the draft I had written at this time), but since then you must have done a modification that had fixed it.

*
But more than this, I ask myself it many other tags/marks/contents/modifications/... associated to a tab would disappear with the same actions (I only tested the 2 mentionned).

*
And also, happy new year everyone! :)

**Comment by madprops** [comment]
Yes color, notes, etc, are stored in the tab's data. Firefox controls this. If you fully close and lose a tab, it's data is gone too. If you want something more permanent, consider using `Domain Rules`, which is a feature Grasshopper has to apply rules to tabs by matching their URL or title.
